c10/cuda/CUDACachingAllocator.h

```diff
@@ -170,8 +170,10 @@ struct TraceEntry {
     SEGMENT_UNMAP, // unmap part of a segment (used with expandable segments)
     SNAPSHOT, // a call to snapshot, used to correlate memory snapshots to trace
               // events
-    OOM // the allocator threw an OutOfMemoryError (addr_ is the amount of free
-        // bytes reported by cuda)
+    OOM, // the allocator threw an OutOfMemoryError (addr_ is the amount of free
+         // bytes reported by cuda)
+    ANNOTATION_START, // API call made to record the start of an annotation
```

Comment:
I'd like to avoid the lists of actions here growing to add additional event types. Can we just have a single 'USER_DEFINED' event here that we use for both annotation start/end, and then put the additional information into gathered context?

Reply:
Changed to USER_DEFINED